nomad/local/grapl-local-infra.nomad

```diff
@@ -142,4 +167,95 @@ job "grapl-local-infra" {
       }
     }
   }
+
+  group "kafka" {
+    network {
+      mode = "bridge"
+    }
+
+    task "kafka" {
+      driver = "docker"
+
+      config {
+        image = "confluentinc/cp-kafka:6.2.0"
+        # Hitting OOM errors
+        memory_hard_limit = 2048
```

Comment:
I'd recommend using the `resources` stanza. By default tasks are allotted 100Mhz compute and 300 MB memory usage. This 300 is probably being passed into the container. By setting the `memory_hard_limit` in this way, I believe (and I could be wrong) that nomad will still treat this task as if it only has 300MB of memory usage which means it could place it on a machine without much headroom left for memory. This could result in an OOM at some point.

Perhaps try 

```
resource {
  cpu = 1000 # or more; we'll need to set resources for all tasks when we get around to deploying in production
  memory = 2048 # this will act as a hard limit and reserve 2GB of memory for this task specifically
}
```